docs/getting-started/tutorials/table_api.md

```diff
@@ -0,0 +1,250 @@
+## Breaking Down The Code
+
+#### The Execution Environment
+
+The first two lines set up our `ExecutionEnvironment`.
+The execution environment is how we set properties for our deployments, specify whether we are writing a batch or streaming application, and create our sources.
+Here we have chosen to use the batch environment since we are building a periodic batch report.
+We then wrap it in a `BatchTableEnvironment` so to have full access to the Table Api.
+
+{% highlight java %}
+ExecutionEnvironment env   = ExecutionEnvironment.getExecutionEnvironment();
+BatchTableEnvironment tEnv = BatchTableEnvironment.create(env);
+{% endhighlight %}
+
+#### Registering Tables
+
+Next, we register tables that we can use for the input and output of our application. 
+The TableEnvironment maintains a catalog of tables that are registered by name. There are two types of tables, input tables and output tables.
+Input tables can be referenced in Table API and SQL queries and provide input data.
+Output tables can be used to emit the result of a Table API or SQL query to an external system.
+Tables can support batch queries, streaming queries, or both. 
+
+{% highlight java %}
+tEnv.registerTableSource("transactions", new TransactionTableSource());
+tEnv.registerTableSink("stdout", new StdOutTableSink());
+{% endhighlight %}
+
+We register two tables, a `transactions` input table and a `stdout` output table. 
+The `transactions` table lets us read credit card transactions, which contain `accountId`'s, `timestamp`'s, and dollar `amount`'s. 
+The `stdout` table writes the output of a job to standard out so we can easily see our results. 
+Both tables support running batch and streaming applications.
+
+#### Registering A UDF
+
+Along with tables we include a user defined function for working with timestamps. Our function takes a timestamp and rounds it down the nearest hour. 
+
+{% highlight java %}
+tEnv.registerFunction("truncateDateToHour", new TruncateDateToHour());
+{% endhighlight %}
+
+#### The Query
+
+With our environment configured and tables registered we are ready to build our first application.
+From the `TableEnvironment` we can `scan` an input table to read its rows and then write those results into an output table using `insertInto`. 
+
+{% highlight java %}
+tEnv
+    .scan("transactions")
+    .insertInto("stdout");
+{% endhighlight %}
+
+This intial job simply reads all transactions and writes them to standard out. 
+
+#### Execute
+
+Flink applications are built lazily and shipped to the cluster for execution only once fully formed. 
+We call `ExecutionEnvironment#execute` to begin the execution of our job. 
+
+{% highlight java %}
+env.execute("Spend Report");
+{% endhighlight %}
+
+## Attempt One
+
+Now that we have the skeleton of a job, let's add some business logic.
+We want a report that shows the total spend for each account across each hour of the day.
+Just like a SQL query, we can select the required fields and group by our keys. 
```

Comment:
an SQL query?

Reply:
Its "a SQL query". The rule is that you use “a" before words that start with a consonant sound and “an” before words that begin with a vowel sound. English is weird.

Reply:
I pronounced it incorrectly :) Like "as kwu al", not sequel.